packages/dev/core/src/Shaders/ShadersInclude/pbrBlockReflection.fx

```diff
@@ -297,7 +305,29 @@
             #else
                 vec4 environmentIrradiance4 = sampleReflection(irradianceSampler, reflectionCoords);
             #endif
-            environmentIrradiance = environmentIrradiance4.rgb;
+            
+            // If we have a predominant light direction, use it to compute the diffuse roughness term.abort
+            // Otherwise, bend the irradiance vector to simulate retro-reflectivity of diffuse roughness.
+            #ifdef USE_IRRADIANCE_DOMINANT_DIRECTION
+                vec3 Ls = normalize(reflectionDominantDirection);
+                float NoL = dot(irradianceVector, Ls);
+                float NoV = dot(irradianceVector, irradianceView);
+                
+                vec3 diffuseRoughnessTerm = vec3(1.0);
+                #if BASE_DIFFUSE_ROUGHNESS_MODEL == 0
+                    float LoV = dot (Ls, irradianceView);
+                    float mag = length(reflectionDominantDirection) * 2.0;
+                    diffuseRoughnessTerm = diffuseBRDF_EON(vec3(1.0), diffuseRoughness, NoL, NoV, LoV) * PI;
+                    diffuseRoughnessTerm = mix(vec3(1.0), diffuseRoughnessTerm, sqrt(min(mag * NoV, 1.0)));
+                #elif BASE_DIFFUSE_ROUGHNESS_MODEL == 1
+                    vec3 H = (irradianceView + Ls)*0.5;
+                    float VoH = dot(irradianceView, H);
+                    diffuseRoughnessTerm = vec3(diffuseBRDF_Burley(NoL, NoV, VoH, diffuseRoughness) * PI);
+                #endif

```

Comment:
Do you have any pointers that would help us understand these calculations?

In the [Frostbite doc (page 70)](https://seblagarde.wordpress.com/wp-content/uploads/2015/07/course_notes_moving_frostbite_to_pbr_v32.pdf), they calculate a dominant direction, but it's used to query the irradiance map, not to alter the result of the query.

Reply:
So, diffuse roughness depends on both the viewing angle and light direction. Here, I'm trying to modify the existing Lambertian prefiltered IBL to approximate EON and Burley using a single light direction for the IBL.
Basically, it has the effect of brightening glancing angles when the light and view vector align.
The magnitude of the dominant direction is kind of a measure of how strongly anisotropic the lighting in the IBL is.